frontend/packages/console-shared/src/components/spotlight/Spotlight.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import cx from 'classnames';
+import Portal from '@console/shared/src/components/popper/Portal';
+import './Spotlight.scss';
+
+type SpotlightProps = {
+  selector: string;
+  animate?: boolean;
+  hasBackdrop?: boolean;
+};
+
+const Spotlight: React.FC<SpotlightProps> = ({ selector, animate, hasBackdrop }) => {
+  const element = document.querySelector(selector);
+  const { height, width, top, left } = element.getBoundingClientRect();
+  const style: React.CSSProperties = {
+    top,
+    left,
+    height,
+    width,
+  };
+  return (
+    <Portal>
+      {hasBackdrop ? (
+        <div className="pf-c-backdrop ocs-spotlight__with-backdrop">
+          <div
+            className={cx({
+              'ocs-spotlight__element-highlight-animate': animate,
+              'ocs-spotlight__element-highlight-noanimate': !animate,
+            })}
+            style={style}
```

Comment:
DRY. Store this component in a variable and render conditionally.

Reply:
updated using variable.